src/renderer/components/swap/Swap.tsx

```diff
@@ -276,6 +275,11 @@ export const Swap = ({
     return max1e8BaseAmount(swapResultAmount)
   }, [swapData.swapResult, targetAssetDecimal])
 
+  const isCausedSlippage = useMemo(() => {
+    const limitFromAmount = amountToSwapMax1e8.amount().multipliedBy((100 - slipTolerance) / 100)
+    return amountToSwapMax1e8.div(bn(100)).times(swapData.slip).gt(amountToSwapMax1e8.minus(limitFromAmount))
+  }, [amountToSwapMax1e8, swapData.slip, slipTolerance])
```

Comment:
Could you extract this function into `Swap.utils` and add a test, pls?

Reply:
Now it's simplified :)